src/eal.cpp

```diff
@@ -2,80 +2,139 @@
 
 #include <json.hpp>
 
+#include <gsl/gsl_interp.h>
 #include <gsl/gsl_spline.h>
 
 #include <string>
 
 using json = nlohmann::json;
+using namespace std;
+
+namespace eal {
+
+  // Parsing the JSON
+  json constructStateFromIsd(const string positionRotationData) {
+     // Parse the position and rotation data from isd
+     json isd = json::parse(positionRotationData);
+     json state;
+
+     state["m_w"] = isd.at("w");
+     state["m_x"] = isd.at("x");
+     state["m_y"] = isd.at("y");
+     state["m_z"] = isd.at("z");
+
+     return state;
+   }
+
+  // Positional Functions
+
+  // Position Data Functions
+  vector<double> getPosition(vector<vector<double>> coords, vector<double> times,
+                             interpolation interp, double time) {
+    // Check that all of the data sizes are okay
+    // TODO is there a cleaner way to do this? We're going to have to do this a lot.
```

Comment:
Could always add a 'validatePosition()` or `'isValidPosition()` function?

Reply:
But that could be in a future PR.

Reply:
That is probably the way to go.